### PR TITLE
static musl builds

### PR DESCRIPTION
**Pull request by jan-guenter**
### Description of changes

_for more details on the reasoning behind these changes see #84_

- added yaml definitions to `.editorconfig` for comfortable workflow editing
- changed static build behavior  to restrict `getpwuid` usage only in static builds with glibc so that static musl build have full functionality
- changed the continuous build workflow to create static musl binaries
- updated Makefile
  - added cf-protection for i*86 builds
  - added platform and arch detection from CXX machine tuple
  - add `PLATFORM_LC`, a lowercase version of `PLATFORM`, for case insensitive comparison
  - added bool flag `STRIP` to force `-s` linker flag
  - turned linker warnings into errors in case of `STATIC` to fail in case of glibc static linking issues
  - bumped version to 1.3
- updated README.md to reflect Makefile changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,21 +14,27 @@ endif
 
 PREFIX ?= /usr/local
 
-#? NOTICE! Manually set PLATFORM and ARCH if not compiling for host system
-PLATFORM ?= $(shell uname -s || echo unknown)
-ARCH ?= $(shell uname -m || echo unknown)
+#? Detect PLATFORM and ARCH from gcc if not set
+PLATFORM ?= $(shell $(CXX) -dumpmachine | awk -F"-" '{ print (NF==4) ? $$3 : $$2 }')
```
Comment (aristocratos) [inline Makefile:18]: This will give you "apple" on OSX it seems like. So line 36 to 41 shouldn't be needed anymore.

File: Makefile [hunk 2]
```diff
@@ -50,13 +56,13 @@ ifeq ($(CXX),g++)
...
-ifeq ($(PLATFORM),Linux)
+ifeq ($(PLATFORM_LC),linux)
 	PLATFORM_DIR := linux
 	THREADS	:= $(shell getconf _NPROCESSORS_ONLN 2>/dev/null || echo 1)
-else ifeq ($(PLATFORM),FreeBSD)
+else ifeq ($(PLATFORM_LC),freebsd)
 	PLATFORM_DIR := freebsd
 	THREADS	:= $(shell getconf NPROCESSORS_ONLN 2>/dev/null || echo 1)
-else ifeq ($(PLATFORM),OSX)
+else ifeq ($(PLATFORM_LC),osx)
```
Comment (aristocratos) [inline Makefile:65]: This should then be "apple" instead of "osx"

File: README.md [hunk 3]
```diff
@@ -259,7 +259,10 @@ Also needs a UTF8 locale and a font that covers:
 
    Append `QUIET=true` for less verbose output.
 
-   Notice! Manually set `$ARCH` variable if cross-compiling
+   Append `STRIP=true` to force stripping of debug symbols (adds `-s` linker flag).
+
+   Append `ARCH=<architecture>` to manually set the target architectur.
```
Comment (aristocratos) [inline README.md:264]: Just a missing "e" :)

File: src/linux/btop_collect.cpp [hunk 4]
```diff
@@ -27,7 +27,7 @@ tab-size = 4
 #include <ifaddrs.h>
 #include <net/if.h>
 
-#ifndef STATIC_BUILD
+#if !defined(STATIC_BUILD) || !defined(__GLIBC__)
```
Comment (aristocratos) [inline src/linux/btop_collect.cpp:30]: ~~Won't this also exclude normal builds (dynamic builds using glibc)?~~

Disregard that, the double negatives got me...

**Review by aristocratos — CHANGES_REQUESTED** [review]

File: src/linux/btop_collect.cpp (hunk 4 above)
Comment (jan-guenter) [reply src/linux/btop_collect.cpp:30]: Should I perhaps make it read:
```cpp
#if !(defined(STATIC_BUILD) && defined(__GLIBC__)
```

File: Makefile (hunk 2 above)
Comment (jan-guenter) [reply Makefile]: changed to apple

File: README.md (hunk 3 above)
Comment (jan-guenter) [reply README.md]: fixed spelling

File: src/linux/btop_collect.cpp (hunk 4 above)
Comment (aristocratos) [reply src/linux/btop_collect.cpp:30]: Sure, I would say that's easier to parse.

**Pull request merged**